src/update_agent/actor.rs

```diff
@@ -188,18 +188,16 @@ impl UpdateAgent {
     fn tick_check_updates(&mut self) -> ResponseActFuture<Self, Result<(), ()>> {
         trace!("trying to check for updates");
 
-        let can_check = self.strategy.can_check_and_fetch(&self.identity);
-        let state_change = actix::fut::wrap_future::<_, Self>(can_check)
-            .then(|can_check, actor, _ctx| actor.local_deployments(can_check))
+        let state_change = actix::fut::ok::<bool, bool, Self>(true)
```

Comment:
This looks like a strange construct, would it be possible to start the chain with `self.local_deployments()` directly? If not, we can at least change the `true` into a `()`, and the `_can_check` into a plain `_`.

(I know future types are a bit overwhelming, so if your answer is "I tried but I'm not sure whether it's possible" I'm fine with just leaving a "TODO" note here to later re-evaluate whether it can be cleaned up further)

Reply:
It turns out a `self.local_deployments()` directly will do it. Thanks a lot for the hint, I got quite confused about this part ..

Reply:
I totally share the sentiment, there is still a huge developer-UX gap in the futures/async area. It's slowly but steadily improving though.